jersey/src/main/java/com/netflix/conductor/server/resources/WorkflowBulkResource.java

```diff
@@ -116,4 +112,19 @@ public BulkResponse retry(List<String> workflowIds) {
     public BulkResponse terminate(List<String> workflowIds, @QueryParam("reason") String reason) {
         return workflowBulkService.terminate(workflowIds, reason);
     }
+
+    /**
+     * Remove workflows for a given correlation id.
+     * @param correlationId - correlationId of the workflows
+     * @param archiveWorkflow - flag to specify whether to archive a workflow, by default true
+     * @return bulk response object containing a list of succeeded workflows and a list of failed ones with errors
+     */
+    @DELETE
+    @Path("/correlationId/{correlationId}")
+    @ApiOperation("Remove workflows for a given correlation id")
+    @Consumes(MediaType.WILDCARD)
+    public BulkResponse removeCorrelatedWorkflows(@PathParam("correlationId") String correlationId,
+                               @QueryParam("archiveWorkflow") @DefaultValue("true") boolean archiveWorkflow) {
```

Comment:
Have you tested to make sure without passing archiveWorkflow, it does get archived in ES? 
Have you tested with the archiveWorkflow query param, explicitily set to true and false, and validation with ES data/